extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/caffeine/CaffeineCacheBuildRecorder.java

```diff
@@ -19,14 +20,15 @@ public void buildCaches(BeanContainer beanContainer, Set<CaffeineCacheInfo> cach
         // The number of caches is known at build time so we can use fixed initialCapacity and loadFactor for the caches map.
         Map<String, CaffeineCache> caches = new HashMap<>(cacheInfos.size() + 1, 1.0F);
 
+        ManagedExecutor managedExecutor = beanContainer.instance(ManagedExecutor.class);
```

Comment:
Hi @mkouba!

Sorry to bother you but I have a small issue with this bean injection which is always `null` at runtime. I've tried many things to fix that, but nothing worked so far.

Is that possible to inject and use `ManagedExecutor` from a `STATIC_INIT` recorder or am I trying to do something silly here? :)

FYI, `ManagedExecutor` is produced by [SmallRyeContextPropagationProvider](https://github.com/quarkusio/quarkus/blob/master/extensions/smallrye-context-propagation/runtime/src/main/java/io/quarkus/smallrye/context/runtime/SmallRyeContextPropagationProvider.java) and registered by [SmallRyeContextPropagationProcessor](https://github.com/quarkusio/quarkus/blob/master/extensions/smallrye-context-propagation/deployment/src/main/java/io/quarkus/smallrye/context/deployment/SmallRyeContextPropagationProcessor.java).

Reply:
No problem at all ;-). 

So you need to be sure the `SmallRyeContextManagerProvider` is already initialized and the `managedExecutor` field is set. In other words, your build step must be executed after the [SmallRyeContextPropagationProcessor#buildStatic()](https://github.com/quarkusio/quarkus/blob/master/extensions/smallrye-context-propagation/deployment/src/main/java/io/quarkus/smallrye/context/deployment/SmallRyeContextPropagationProcessor.java#L37-L69).

Ideally, the `SmallRyeContextPropagationProcessor` should produce a build item similar to `io.quarkus.deployment.builditem.ExecutorBuildItem` so that you can simply inject this build item, and possibly pass the `ManagedExecutor` reference to your recorder.